streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1147,6 +1162,10 @@ public String toString(final String indent) {
         return indent + "\tStreamsThread threadId: " + getName() + "\n" + taskManager.toString(indent);
     }
 
+    public String getGroupInstanceID(){
+        return mainConsumer.groupMetadata().groupInstanceId().orElse("");
```

Comment:
It seems easier to get it form here than the config. It looked like I might have how to manipulate strings in that case

Reply:
I would do it the same way.